packages/core/components/DirectoryTree/findChildNodes.ts

```diff
@@ -0,0 +1,102 @@
+import { defaults, isEmpty, pull } from "lodash";
+
+import { NO_VALUE_NODE, ROOT_NODE } from "./directory-hierarchy-state";
+import FileSet from "../../entity/FileSet";
+import ExcludeFilter from "../../entity/FileFilter/ExcludeFilter";
+import { AnnotationService, FileService } from "../../services";
+import { naturalComparator } from "../../util/strings";
+
+export interface FindChildNodesParams {
+    ancestorNodes?: string[];
+    currentNode: string;
+    hierarchy: string[];
+    annotationService: AnnotationService;
+    fileService: FileService;
+    fileSet: FileSet;
+    shouldShowNullGroups?: boolean;
+}
+
+const DEFAULTS = {
+    ancestorNodes: [],
+    currentNode: ROOT_NODE,
+    shouldShowNullGroups: false,
+};
+
+export async function findChildNodes(params: FindChildNodesParams): Promise<string[]> {
+    const {
+        ancestorNodes,
+        currentNode,
+        fileSet,
+        hierarchy,
+        annotationService,
+        fileService,
+        shouldShowNullGroups,
+    } = defaults({}, params, DEFAULTS);
+
+    const isRoot = currentNode === ROOT_NODE;
```

Comment:
This looked funny to me, maybe just separate `(currentNode === ROOT_NODE)`

Reply:
I tried adding the parentheses in [a43d4e1](https://github.com/AllenInstitute/biofile-finder/pull/498/commits/a43d4e1a4905481ad5cec33b4798d0d714516125) but the linter removed them when I pushed